File: day01/day1.py
def part1(nums):
    return sum(1 for l, r in zip(nums, nums[1:]) if l < r)

def part2(nums):
    return part1([nums[i]+nums[i+1]+nums[i+2] for i in range(len(nums)-2)])

def main():
    with open('input.txt') as f:
        nums = [int(n) for n in f.readlines() if len(n.strip()) > 0]

    print(f'Part 1: {part1(nums)}')
    print(f'Part 2: {part2(nums)}')

if __name__ == "__main__":
    main()

File: day01/day1.cpp
#include <iostream>
#include <fstream>
#include <string>

#define INPUT_FILE "input.txt"
#define N 2000

int n = 0;
int data[N];

int part1() {
    int ans = 0;
    for (int i = 1; i < n; i++) {
        if (data[i-1] < data[i]) ans++;
    }
    return ans;
}

int part2() {
    int ans = 0;
    for (int i = 3; i < n; i++) {
        if (data[i-3] + data[i-2] + data[i-1] < data[i-2] + data[i-1] + data[i])
            ans++;
    }
    return ans;
}

void read_data() {
    std::ifstream stream(INPUT_FILE);
    std::string line;
    while (std::getline(stream, line))
        data[n++] = std::stoi(line);
}

int main() {
    read_data();
    std::cout << "Part 1: " << part1() << std::endl;
    std::cout << "Part 2: " << part2() << std::endl;
}